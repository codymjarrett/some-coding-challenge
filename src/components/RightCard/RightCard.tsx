import React from 'react'

export interface RightCardInputProps {
	movie: any
}

const RightCard: React.SFC<RightCardInputProps> = ({ movie }) => (
	<div className="rightCard">
		<p className="detail-highlight">
			Location{movie.locations.length > 1 ? 's' : ''}
		</p>
		{movie.locations.length > 0 ? (
			<ul>
				{movie.locations.map((loc: any, index: number) => (
					<li className="rightCard__detail" key={index}>
						{loc}
					</li>
				))}
			</ul>
		) : (
			<p className="rightCard">No disclosed Location</p>
		)}
	</div>
)

export default RightCard
