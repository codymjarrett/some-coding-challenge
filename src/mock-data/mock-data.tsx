export const mockData = () => {
	return {
		title: "Alexander's Ragtime Band",
		release_year: '1938',
		locations: [
			'Cliff House (1090 Point Lobos Avenue)',
			'The Walden House, Buena Vista Park',
		],
		fun_facts:
			'In 1887, the Cliff House was severely damaged when the schooner Parallel, abandoned and loaded with dynamite, ran aground on the rocks below.',
		production_company: 'Twentieth Century Fox Film Corporation',
		distributor: 'Twentieth Century Fox Film Corporation',
		director: 'Henry King',
		writer: 'Kathryn Scola',
		actor_1: 'Tyrone Power',
		actor_2: 'Alice Faye',
		actor_3: 'Don Ameche',
		actors: ['Tyrone Power', 'Alice Faye', 'Don Ameche'],
	}
}
