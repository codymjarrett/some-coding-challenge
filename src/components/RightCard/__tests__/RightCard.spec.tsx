import React from 'react'
import { render, cleanup } from '@testing-library/react';
import { mockData } from '../../../mock-data/mock-data'

import RightCard from '../RightCard'

afterEach(cleanup);

it("renders", () => {
  const { asFragment } = render(<RightCard movie={mockData()}/>);
  expect(asFragment()).toMatchSnapshot();
});

