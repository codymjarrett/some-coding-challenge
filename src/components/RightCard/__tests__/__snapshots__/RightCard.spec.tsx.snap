// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="rightCard"
  >
    <p
      class="detail-highlight"
    >
      Locations
    </p>
    <ul>
      <li
        class="rightCard__detail"
      >
        Cliff House (1090 Point Lobos Avenue)
      </li>
      <li
        class="rightCard__detail"
      >
        The Walden House, Buena Vista Park
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
